packages/headless/src/api/search/search-api-client.ts

```diff
@@ -183,7 +183,12 @@ export class SearchAPIClient {
     const body = await response.json();
 
     if (isSuccessSearchResponse(body)) {
-      return {success: body};
+      const processedResponse =
+        await this.options.postprocessSearchResponseMiddleware({
+          response,
+          body,
+        });
+      return {success: processedResponse.body};
```

Comment:
I feel like it would be convenient if we also supported void return types to allow something like this:
```js
{
  postprocessSearchResponseMiddleware: (response) => {
    response.body.results.forEach(result => result.raw.abc = 'def');
  },
}
```

```suggestion
      const params = {
        response,
        body,
      };
      const processedResponse =
        await this.options.postprocessSearchResponseMiddleware(params) ?? params;
      return {success: processedResponse.body};
```